### PR TITLE
[CST] Override the display name and status of "PMR Pending" requests.

### PR DESCRIPTION
**Pull request by LouisFettet**
## Summary

The change here for the end-user is very simple:
1. Instead of showing PMR Pending requests as first-party requests, we show them as third-party requests.
2. Instead of labeling these requests "PMR Pending", we show "Private Medical Record".

The changes under the hood are:
* Clean up and add transform logic to the `serialize` reducer.
  * Move around some of the existing code in the reducer so that all the logic in the `serializeClaim` function uses other functions. These changes should be aesthetic only and shouldn't impact the result of the function.
  * Replace display names _upon receiving them from the Lighthouse API, before they're ever passed to containers or components_.
  * Replace the status of PMR Pending requests in the same way. This logic is wrapped in a `<tempfix>` comment block to make it clear that this is not a permanent solution to the problem.  
* Create a helper function to replace the display names of tracked items. Currently it only supports replacing "PMR Pending" with "Private Medical Record", but we know there are likely some other types of requests that we'll want to make more human-readable in the future as well.
* Added tests to cover all changes.

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/93456

## Testing done

@jerekshoe added a PMR Pending tracked item on claim `600555011` on user `23` (Kyle Cole) in staging.

- On the Claim Details page:
  - On the Status tab:
    - There should no longer be a "warning" alert under "What you need to do"
    - The tracked item under recent activity should show an "information" alert and use the new display name.
  - On the Files tab, the "warning" alert should be replaced with an "information" alert and the new display name is used.
- On the Document Request page, the "optional" text is rendered in the description and the new display name is used.

It's possible that this tracked item appears on other pages that I'm unaware of, but the idea is that it should render properly as a third-party request with the updated display name across the app (since we're modifying it as soon as we receive data from the Lighthouse API).

A search for `PMR Pending` of the app returned no results, so it doesn't seem like there is any existing logic depending on the original display name or status (which should make this a safe change).

## Screenshots

|         | Before | After |
| ------- | ------ | ----- |
| Claim status - What you need to do  | Screenshot 2024-10-03 at 1 35 38 PM |  Screenshot 2024-10-03 at 1 36 03 PM     |
| Claim status - Recent activity  |  Screenshot 2024-10-03 at 1 36 40 PM      |  Screenshot 2024-10-03 at 1 37 09 PM     |
| Claim files - Additional evidence | Screenshot 2024-10-03 at 2 18 09 PM | Screenshot 2024-10-03 at 2 18 45 PM |
| Document request | Screenshot 2024-10-03 at 1 37 41 PM | Screenshot 2024-10-03 at 1 38 13 PM |

## What areas of the site does it impact?

Claim Status Tool

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [x] Documentation has been updated
  - Not necessary.
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed
  - Not necessary as only the data has changed (not any of the rendering logic).

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
  - Not necessary.
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)
  - Not necessary.

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user
  - Not necessary.

## Requested Feedback

See self-review!

### CODE REVIEW

File: src/applications/claims-status/reducers/serialize.js [hunk 1]
```diff
@@ -25,27 +29,49 @@ const getDocsAssociatedWithTrackedItems = docs =>
...
 };
 
 const associateDocsWithTrackedItems = (items, docs) => {
-  return items.map(item => {
-    const newItem = { ...item };
-    const associatedDocs = getDocsAssociatedWithTrackedItem(newItem, docs);
-    newItem.documents = associatedDocs || [];
-    newItem.date = getTrackedItemDate(item);
-    return newItem;
-  });
```
Comment (LouisFettet) [inline src/applications/claims-status/reducers/serialize.js:39]: Cleaned this up a little to make it more concise. Notable changes are:
1. I removed the `|| []` for assigning `documents`, since I don't believe we can get here with a falsy return value for `getDocsAssoociatedWithTrackedItems` (since `docs` need to be an array already to run `filter` on them). For additional safety, I went ahead and added a default value (empty array) for both `supportingDocuments` and `trackedItems` at the beginning of the `serializeClaim` function.
2. I moved setting `date` on the tracked item into the transform function, as I don't believe it's related to associating docs with the tracked item.

File: src/applications/claims-status/reducers/serialize.js [hunk 2]
```diff
@@ -25,27 +29,49 @@ const getDocsAssociatedWithTrackedItems = docs =>
...
 };
 
 const filterAssociatedDocs = docs =>
   docs.filter(doc => !isAssociatedWithAnyTrackedItem(doc));
 
+const transformUnassociatedDocs = docs =>
+  docs.map(doc => ({
+    ...doc,
+    date: doc.uploadDate,
+  }));
```
Comment (LouisFettet) [inline src/applications/claims-status/reducers/serialize.js:49]: This was moved into its own function from the original return of `serializeClaim`.

File: src/applications/claims-status/reducers/serialize.js [hunk 3]
```diff
@@ -25,27 +29,49 @@ const getDocsAssociatedWithTrackedItems = docs =>
...
+
+const transformAssociatedTrackedItems = items =>
+  items.map(item => ({
+    ...item,
+    // <tempfix>
```
Comment (LouisFettet) [inline src/applications/claims-status/reducers/serialize.js:54]: Do we have an established pattern I should use here for marking code as temporary? Ideally this would be some sort of consistent token we can search for during clean-up work.

File: src/applications/claims-status/reducers/serialize.js [hunk 4]
```diff
@@ -56,19 +82,20 @@ export const serializeClaim = claim => {
...
-        doc => {
-          return {
-            ...doc,
-            date: doc.uploadDate,
-          };
-        },
-      ),
-      trackedItems: associatedTrackedItems,
+      supportingDocuments: transformedUnassociatedDocuments,
+      trackedItems: transformedAssociatedTrackedItems,
```
Comment (LouisFettet) [inline src/applications/claims-status/reducers/serialize.js:98]: These variables are a bit wordy; if we don't find the labeling helpful, we can just run the relevant functions like so:
```js
      supportingDocuments: transformUnassociatedDocs(
        filterAssociatedDocs(supportingDocuments),
      ),
      trackedItems: transformAssociatedTrackedItems(
        associatedTrackedItems,
      ),
```

The result is the same regardless.

File: src/applications/claims-status/tests/reducers/serialize.unit.spec.jsx [hunk 5]
```diff
@@ -136,4 +152,48 @@ describe('Claim Serializer', () => {
...
+          },
+        ],
+      },
+    };
+    const serializedClaim = serializeClaim(claim);
+    const trackedItem1 = serializedClaim.attributes.trackedItems.find(
+      d => d.id === 1,
+    );
+    expect(trackedItem1.displayName).to.eql('Private Medical Record');
+  });
```
Comment (LouisFettet) [inline src/applications/claims-status/tests/reducers/serialize.unit.spec.jsx:198]: These tests are basically doing the same thing, but one tests the `status` while the other tests the `displayName`. Since testing the status is temporary, I kept them separate so that we can just remove that whole test case down the road.

**Review by LouisFettet — COMMENTED** [review]
Self-review!

**Comment by jacobworrell** [comment]
@LouisFettet looks good!

**Review by rmessina1010 — APPROVED** [review]
LGTM

**Review by jerekshoe — APPROVED** [review]
